src/mixins/filing-template-mixin.ts

```diff
@@ -300,7 +300,10 @@ export default class FilingTemplateMixin extends Mixins(AmalgamationMixin, DateM
     }
 
     // restore the Amalgamation Court Approval if it's True or False
-    if (draftFiling.amalgamationApplication.courtApproval !== null) {
```

Comment:
This code failed if courtApproval did not exist on the amalgamationApplication object (eg, initial draft).

Because courtApproval was undefined (not null), line 304 was run, which saved "undefined" in the store. Then, in AmalgamationStatement.vue, [this code](https://github.com/bcgov/business-create-ui/blob/467fd3e11f512392543b6413a7cbaf4d63428a22/src/components/Amalgamation/AmalgamationStatement.vue#L80) set the validity to True (which is incorrect), and the overall filing validation did not catch that the amalgamation statement had not been selected.

Now, it all works correctly.